src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,105 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * This class is a subclass of TickingTimer. It contains logic to facilitate dynamic period of the timer.
+ */
+public class RefreshTimer extends TickingTimer{
+
+    /**
+     * The buffer time to be added to ensure refresh happens after the api quota renewal.
+     */
+    public static final int BUFFER_TIME = 1;
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    public static final int APIQUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period of the refresh rate.
+     */
+    public static final int DEFAULT_REFRESH_PERIOD_IN_MIN = 1;
+
+    public RefreshTimer(String name, int period, Consumer<Integer> onTick, Runnable onTimeout,
+                        TimeUnit timeUnit) {
+        super(name, period, onTick, onTimeout, timeUnit);
+    }
+
+    /**
+     * Computes the TickerTimer period that are used for refreshing the issues periodically.
+     * The refresh rate is computed based on the three input argument and constant APIQUOTA_BUFFER.
+     * APIQUOTA_BUFFER is the amount that is reserved for manual operations performed by the user.
+     * This amount is then deducted from the actual apiQuota to ensure the algorithm reserves this amount.
+     * The calculation has 2 steps.
+     * The first step, the offset amount(apiQuota - APIQUOTA_BUFFER) is divided by the amount
+     * of api used during the last refresh. This amount corresponds to the estimated no of refresh allowed to use till
+     * the next apiQuota renewal.
+     * Second, the refresh rate is calculated by dividing the remainingTimeInMin by the estimated no of refresh allowed.
```

Comment:
Rather than the mechanism, explain the intention. The mechanism should be clear from the code. The intention is, assuming future refreshes will take the same number of API calls as the last refresh, find out the refresh duration that will spread out the refreshes until the next quota top up.